source/code/plugin/filter_inventory2mdm.rb

```diff
@@ -0,0 +1,261 @@
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+        @@pod_inventory_custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "insights.container/nodes", 
+                        "dimNames": [ 
+                        "phase", 
+                        "namespace", 
+                        "node", 
+                        "controllerName"
+                        ], 
+                        "series": [ 
+                        { 
+                            "dimValues": [ 
+                            "%{phaseDimValue}", 
+                            "%{namespaceDimValue}", 
+                            "%{nodeDimValue}", 
+                            "%{controllerNameDimValue}"
+                            ], 
+                            "min": %{podCountMetricValue},
+                            "max": %{podCountMetricValue}, 
+                            "sum": %{podCountMetricValue}, 
+                            "count": 1 
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+        @process_incoming_stream = true
+
+		def initialize
+            super
+		end
+
+		def configure(conf)
+			super
+			@log = nil
+			
+			if @enable_log
+				@log = Logger.new(@log_path, 'weekly')
+				@log.debug {'Starting filter_inventory2mdm plugin'}
+			end
+		end
+
+        def start
+            super
+            @process_incoming_stream = check_custom_metrics_availability
+            @log.debug "After check_custom_metrics_availability process_incoming_stream #{@process_incoming_stream}"
+        end
+
+        def check_custom_metrics_availability
+            aks_region = ENV['AKS_REGION']
+            aks_resource_id = ENV['AKS_RESOURCE_ID']
+            if aks_region.to_s.empty? && aks_resource_id.to_s.empty?
+                return false # This will also take care of AKS-Engine Scenario. AKS_REGION/AKS_RESOURCE_ID is not set for AKS-Engine. Only ACS_RESOURCE_NAME is set
+            end
+            @log.debug "AKS_REGION #{aks_region}"
+            custom_metrics_regions_arr = @custom_metrics_azure_regions.split(',')
+            custom_metrics_regions_hash = custom_metrics_regions_arr.map {|x| [x.downcase,true]}.to_h
+
+            @log.debug "Custom Metrics Regions Hash #{custom_metrics_regions_hash}"
+
+            if custom_metrics_regions_hash.key?(aks_region.downcase)
+                @log.debug "Returning true for check_custom_metrics_availability"
+                return true
+            else 
+                @log.debug "Returning false for check_custom_metrics_availability"
+                return false
+            end
+        end
+
+		def shutdown
+			super
+        end
+        
+        # KUBE_NODE_INVENTORY
```

Comment:
Do we need this payload ? Also it has subscription info (which will be open to all in github)